### PR TITLE
feat: added workflow and example

### PR DESCRIPTION
**Pull request by capri-xiyue**
added workflow and example

### CODE REVIEW

File: .github/workflows/mapping-data-validation.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}-mapping-data-validation'
+  cancel-in-progress: true
+
+jobs:
+  snapshot:
+    uses: 'abcxyz/pmap/.github/workflows/resource-mapping-check.yml@v0.0.5-rc' # Change here to pin the tag or commit.
+    with:
+      resource_mapping_directory: 'example_org/mapping' #Change here if you change the folder to store the mappings.
```
Comment (yolocs) [inline .github/workflows/mapping-data-validation.yml]: nit: be consistent
```suggestion
      resource_mapping_directory: 'example_org/mapping' # Change here if you change the folder to store the mappings.
```

File: .github/workflows/mapping-data-validation.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      go_version: '>=1.20.0'
```
Comment (yolocs) [inline .github/workflows/mapping-data-validation.yml]: Should we use 1.21

File: .github/workflows/snapshot-mapping-data-copy.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  snapshot:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@v0.0.5-rc' # Change here to pin the tag or commit.
+    with:
+      workload_identity_provider: '${{ vars.workload_identity_provider }}'
+      service_account: '${{ vars.service_account }}'
+      destination_prefix: '${{ vars.gcs_bucket }}/mapping'
+      path: 'example_org/mapping' #Change here if you change the folder to store the mappings.
```
Comment (yolocs) [inline .github/workflows/snapshot-mapping-data-copy.yml]: The workflow name says it's for policy but here says "mapping"

File: example_org/mapping/product_team_a/product_x/gcs_bucket.yaml
Comment (yolocs) [inline example_org/mapping/product_team_a/product_x/gcs_bucket.yaml:1]: For simplicity, I suggest removing "product_team_x" in the path and only use product or service for grouping

File: example_org/mapping/product_team_a/product_x/gcs_bucket.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+resource:
+  name: '//storage.googleapis.com/pmap-template-static-bucket-23b4'
+  provider: 'gcp'
+  subscope: 'parent/foo/child/bar?key1=value1&key2=value2'
+annotations:
+  wipeout_plan_id: 'policy/product_team_a/product_x_user_data_retention'
+contacts:
+  email:
+    - 'product_team_a@gmail.com'
```
Comment (yolocs) [inline example_org/mapping/product_team_a/product_x/gcs_bucket.yaml]: Use `@example.com` throughout

File: example_org/policy/product_team_a/product_x_user_data_retention.yaml
Comment (yolocs) [inline example_org/policy/product_team_a/product_x_user_data_retention.yaml:1]: Again, for simplicity, I suggest only having one wipeout plan E.g. `policy/wipeout/default.yaml` and both mappings reference it

**Review by yolocs — CHANGES_REQUESTED** [review]

File: .github/workflows/mapping-data-validation.yml (hunk 1 above)
Comment (capri-xiyue) [reply .github/workflows/mapping-data-validation.yml]: done

File: .github/workflows/mapping-data-validation.yml (hunk 2 above)
Comment (capri-xiyue) [reply .github/workflows/mapping-data-validation.yml]: done

File: .github/workflows/snapshot-mapping-data-copy.yml (hunk 3 above)
Comment (capri-xiyue) [reply .github/workflows/snapshot-mapping-data-copy.yml]: I changed the workflow name.

File: example_org/mapping/product_team_a/product_x/gcs_bucket.yaml
Comment (capri-xiyue) [reply example_org/mapping/product_team_a/product_x/gcs_bucket.yaml:1]: done

File: .github/workflows/snapshot-mapping-data-change.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  snapshot:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-change.yml@v0.0.5-rc' # Change here to pin the tag or commit.
+    with:
+      workload_identity_provider: '${{ vars.workload_identity_provider }}'
+      service_account: '${{ vars.service_account }}'
+      destination_prefix: '${{ vars.gcs_bucket }}/mapping'
+      path: 'example_org/mapping' #Change here if you change the folder to store the mappings.
```
Comment (sqin2019) [inline .github/workflows/snapshot-mapping-data-change.yml]: Should we make this repo variable?

Comment (capri-xiyue) [reply .github/workflows/snapshot-mapping-data-change.yml]: I don't think we need it here as users may need to change the path much more frequent compared to the infra setting as wif or wif sa or gcs bucket.

Comment (sqin2019) [reply .github/workflows/snapshot-mapping-data-change.yml]: I am thinking differently, it feels like a repo set up, and should not be changed frequently.

Comment (capri-xiyue) [reply .github/workflows/snapshot-mapping-data-change.yml]: In addition, making it a env variable here makes it more difficult for users to understand what the path is for and users don't have the permissions to see the env variables we set in our github repo.

Comment (capri-xiyue) [reply .github/workflows/snapshot-mapping-data-change.yml]: discussed offline. It is fine to keep it as it is now.

File: example_org/policy/wipeout/default.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+plan_id: policy/wipeout/default
```
Comment (yolocs) [inline example_org/policy/wipeout/default.yaml:1]: Did we agree to use path as the id?

**Review by yolocs — APPROVED** [review]

File: example_org/policy/wipeout/default.yaml (hunk 6 above)
Comment (capri-xiyue) [reply example_org/policy/wipeout/default.yaml:1]: It's part of the design doc. Using path as the id is recommended.

**Review by sqin2019 — APPROVED** [review]

**Review by sqin2019 — APPROVED** [review]

**Review by yolocs — DISMISSED** [review]

**Review by sqin2019 — APPROVED** [review]

**Pull request merged**